src/shared/utilities/childProcess.ts

```diff
@@ -79,8 +79,8 @@ export class ChildProcess {
     private makeResult(code: number, signal?: string): ChildProcessResult {
         return {
             exitCode: code,
-            stdout: this.stdoutChunks.join().trim(),
-            stderr: this.stderrChunks.join().trim(),
+            stdout: this.stdoutChunks.join('').trim(),
+            stderr: this.stderrChunks.join('').trim(),
```

Comment:
this should be a separate commit. the other commits can be squashed into 1 commit